src/ray/raylet/scheduling_resources.h

```diff
@@ -0,0 +1,140 @@
+#ifndef SCHEDULING_RESOURCES_H
+#define SCHEDULING_RESOURCES_H
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+#include "ray/raylet/worker_pool.h"
+
+namespace ray {
+
+/// Resource availability status reports whether the resource requirement is
+/// (1) infeasible, (2) feasible but currently unavailable, or (3) available.
+typedef enum {
+  kInfeasible,          ///< Cannot ever satisfy resource requirements.
+  kResourcesUnavailable,///< Feasible, but not currently available.
+  kFeasible             ///< Feasible and currently available.
+} ResourceAvailabilityStatus;
+
+/// \class ResourceSet
+/// \brief Encapsulates and operates on a set of resources, including CPUs,
+/// GPUs, and custom labels.
+class ResourceSet {
+public:
+
+  /// \brief empty ResourceSet constructor.
+  ResourceSet();
+
+  /// \brief Constructs ResourceSet from the specified resource map.
+  ResourceSet(const std::unordered_map<std::string, double> &resource_map);
+
+  /// \brief Empty ResourceSet destructor.
+  ~ResourceSet();
+
+  /// \brief Test equality with the other specified ResourceSet object.
+  /// \param rhs: Right-hand side object for equality comparison.
+  /// \return True if objects are equal, False otherwise.
+  bool operator==(const ResourceSet &rhs) const;
+
+  /// \brief Test equality with the other specified ResourceSet object.
+  /// \param other: Right-hand side object for equality comparison.
+  /// \return True if objects are equal, False otherwise.
+  bool IsEqual(const ResourceSet &other) const;
+
+  /// \brief Test whether this ResourceSet is a subset of the other ResourceSet.
+  /// \param other: The resource set we check being a subset of.
+  /// \return True if the current resource set is the subset of other. False
+  ///          otherwise.
+  bool isSubset(const ResourceSet &other) const;
```

Comment:
Inconsistent capitalization (this applies in many places)

Reply:
caps should be addressed everywhere now.